napari/plugins/_plugin_manager.py

```diff
@@ -436,3 +440,56 @@ def get_widget(
                 raise KeyError(msg)
 
         return plg_wdgs[widget_name]
+
+    def get_reader_for_extension(self, extension: str) -> Optional[str]:
+        """Get reader plugin assigned to `extension`."""
+        return self._get_plugin_for_extension(extension, type='reader')
+
+    def assign_reader_to_extensions(
+        self, reader: str, *extensions: str
+    ) -> None:
+        """Assign a specific reader plugin to `extension`."""
+        self._assign_plugin_to_extensions(reader, *extensions, type='reader')
+
+    def get_writer_for_extension(self, extension: str) -> Optional[str]:
+        """Get writer plugin assigned to `extension`."""
+        return self._get_plugin_for_extension(extension, type='writer')
+
+    def assign_writer_to_extensions(
+        self, writer: str, *extensions: str
+    ) -> None:
+        """Assign a specific writer plugin to `extension`."""
+        self._assign_plugin_to_extensions(writer, *extensions, type='writer')
+
+    def _get_plugin_for_extension(
+        self, extension: str, type: str
+    ) -> Optional[str]:
+        ext_map = getattr(self, f'_extension2{type}', None)
+        if ext_map is None:
+            raise ValueError(f"invalid plugin type: {type!r}")
+        plugin = ext_map.get(extension)
+        # make sure it's still an active plugin
+        if plugin and (plugin not in self.plugins):
+            del self.ext_map[plugin]
+            return None
+        return plugin
+
+    def _assign_plugin_to_extensions(
+        self, plugin: str, *extensions, type: str = None
+    ) -> None:
+        caller: HookCaller = getattr(self.hook, f'napari_get_{type}', None)
+        if caller is None:
+            raise ValueError(f"invalid plugin type: {type!r}")
+
+        plugins = caller.get_hookimpls()
+        if plugin not in {p.plugin_name for p in plugins}:
+            msg = trans._(
+                "{plugin!r} is not a valid plugin plugin name",
+                plugin=plugin,
+                deferred=True,
+            )
+            raise ValueError(msg)
+
+        ext_map = getattr(self, f'_extension2{type}', {})
+        for ext in extensions:
```

Comment:
@tlambert03 should we check that the extensions are valid extensions supported by napari?

Trying to  register a typo or whatever should probably fail

Reply:
Good point.  Tricky though! Given the lack of static extension support metadata 😖 ...

Not all reader validators are as simple as just checking an extension.  Some (like my ndtiffs extension) do actually need to do a little inspection of the path first.

Reply:
> Good point. Tricky though! Given the lack of static extension support metadata 😖 ...

Hmmm 🤔 

We could at least raise a warning if the extension is not among https://github.com/napari/napari/blob/255822fb17f25c79133ec40dba0e626b2bf66b20/napari/utils/io.py#L38 ?

Reply:
I added something [here](https://github.com/napari/napari/pull/2695/files#diff-666c2641af25b7258430403de4c9c0f695707b008f79c8f1e25c6cbb534d3120R533-R547)... we can always remove it if it turns out to be too "chatty" with false warnings.

Reply:
Sweet!